File: web/src/types/AIChat.ts
export interface AIConversationMessage {
  id: string;
  conversation_id: string;
  user_id: number;
  role: "user" | "assistant";
  content: string;
  created_at: string;
}

export interface AIConversationSummary {
  id: string;
  title: string;
  is_pinned: boolean;
  created_at: string;
  updated_at: string;
  last_message?: string | null;
}

export interface AIConversation {
  id: string;
  title: string;
  is_pinned: boolean;
  created_at: string;
  updated_at: string;
  messages: AIConversationMessage[];
}

export interface AIChatRequest {
  conversation_id?: string | null;
  message: string;
  context?: string | null;
  system_prompt?: string | null;
  is_temporary?: boolean;
}

export interface AIChatResponse {
  id: string | number;
  conversation_id: string;
  message: string;
  response: string;
  title: string;
  user_message?: AIConversationMessage;
  assistant_message?: AIConversationMessage;
  created_at: string;
}

export interface AIChatMessage {
  id: string | number;
  user_id?: number;
  message: string;
  response: string;
  created_at: string;
}

export interface AIHistoryResponse {
  chats: AIChatMessage[];
  total: number;
}

File: backend/schemas/ai_chat.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class AIConversationMessageSchema(BaseModel):
    id: str
    conversation_id: str
    user_id: int
    role: str  # "user" | "assistant"
    content: str
    created_at: datetime

    class Config:
        from_attributes = True


class AIConversationSummarySchema(BaseModel):
    id: str
    title: str
    is_pinned: bool
    created_at: datetime
    updated_at: datetime
    last_message: str | None = None

    class Config:
        from_attributes = True


class AIConversationSchema(BaseModel):
    id: str
    title: str
    is_pinned: bool
    created_at: datetime
    updated_at: datetime
    messages: list[AIConversationMessageSchema] = []

    class Config:
        from_attributes = True


class AIConversationCreate(BaseModel):
    id: str | None = None
    title: str | None = "New Conversation"


class AIConversationUpdate(BaseModel):
    title: str | None = None
    is_pinned: bool | None = None


class AIChatRequest(BaseModel):
    conversation_id: str | None = None
    message: str
    context: str | None = None
    system_prompt: str | None = None
    is_temporary: bool = False
    client_message_id: str | None = None


class AIChatResponse(BaseModel):
    id: str | int | None = None
    conversation_id: str
    message: str
    response: str
    title: str = "New Conversation"
    user_message: AIConversationMessageSchema | None = None
    assistant_message: AIConversationMessageSchema | None = None
    created_at: datetime

    class Config:
        from_attributes = True


class AIChatHistoryResponse(BaseModel):
    chats: list[AIChatResponse]
    total: int
